theano/tensor/nnet/abstract_conv.py

```diff
@@ -177,6 +177,187 @@ def conv2d(input,
     return conv_op(input, filters)
 
 
+def deconv(input,
+           filters,
+           input_shape=None,
+           output_shape=None,
+           filter_shape=None,
+           border_mode='valid',
+           subsample=(1, 1),
+           filter_flip=True):
+    """This function will build the symbolic graph for deconvoloving a
+    mini-batch of a stack of 2D inputs with a set of 2D filters (using
+    gradient of convolution), such that it upsamples the input to the
+    desired output resolution.
```

Comment:
I think it's officially quite wrong to call it "deconv", as deconvolution
is something completely different (namely, trying to reverse the effect of
a convolution to recover the original signal).

On Fri, Jan 15, 2016 at 5:48 PM, Francesco <notifications@github.com> wrote:

> In theano/tensor/nnet/abstract_conv.py
> <https://github.com/Theano/Theano/pull/3872#discussion_r49914640>:
>
> > @@ -177,6 +177,187 @@ def conv2d(input,
> >      return conv_op(input, filters)
> >
> >
> > +def deconv(input,
> > +           filters,
> > +           input_shape=None,
> > +           output_shape=None,
> > +           filter_shape=None,
> > +           border_mode='valid',
> > +           subsample=(1, 1),
> > +           filter_flip=True):
> > +    """This function will build the symbolic graph for deconvoloving a
> > +    mini-batch of a stack of 2D inputs with a set of 2D filters (using
> > +    gradient of convolution), such that it upsamples the input to the
> > +    desired output resolution.
>
> We struggled to find a reference for this when we wrote the paper, but we
> did find a couple of articles. If you want to cite some literature and/or
> use our description of the operation, if you like it, here is the paper
> (ref. to sec. 2.2.4): http://arxiv.org/pdf/1511.07053v2.pdf
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3872/files#r49914640>.
>